x/ecocredit/client/testsuite/grpc.go

```diff
@@ -543,90 +402,50 @@ func (s *IntegrationTestSuite) TestGetSellOrdersByAddress() {
 		tc := tc
 		s.Run(tc.name, func() {
 			resp, err := rest.GetRequest(tc.url)
-			require.NoError(err)
-
-			var sellOrders ecocredit.QuerySellOrdersByAddressResponse
-			err = val.ClientCtx.Codec.UnmarshalJSON(resp, &sellOrders)
-
-			if tc.expErr {
-				require.Error(err)
-				require.Contains(string(resp), tc.errMsg)
+			require.NoError(err, err)
+			var res marketplace.QuerySellOrdersByBatchDenomResponse
+			err = val.ClientCtx.Codec.UnmarshalJSON(resp, &res)
+			require.NoError(err, err)
+			require.NotNil(res.SellOrders)
+			if tc.expItems > 0 {
+				require.Len(res.SellOrders, tc.expItems)
 			} else {
-				require.NoError(err)
-				require.NotNil(sellOrders.SellOrders)
-				require.Len(sellOrders.SellOrders, tc.expItems)
+				require.Greater(len(res.SellOrders), 0)
```

Comment:
Same as above. Not sure why we are using `-1` for expected number of items.

Reply:
see above